pkg/controller/volume/expand/expand_controller_test.go

```diff
@@ -0,0 +1,182 @@
+			hasError: true,
+		},
+		{
+			name:   "when pvc has no storageclass",
+			pv:     getFakePersistentVolume("vol-1", csitranslationplugins.AWSEBSInTreePluginName, "no-sc-pvc-vol-1"),
+			pvc:    getFakePersistentVolumeClaim("no-sc-pvc", "vol-1", "", "no-sc-pvc-vol-1"),
+			pvcKey: "default/no-sc-pvc",
+		},
+		{
+			name:   "when pvc storageclass is missing",
+			pv:     getFakePersistentVolume("vol-2", csitranslationplugins.AWSEBSInTreePluginName, "missing-sc-pvc-vol-2"),
+			pvc:    getFakePersistentVolumeClaim("missing-sc-pvc", "vol-2", "resizable", "missing-sc-pvc-vol-2"),
+			pvcKey: "default/missing-sc-pvc",
+		},
+		{
+			name:            "when pvc and pv has everything for in-tree plugin",
+			pv:              getFakePersistentVolume("vol-3", csitranslationplugins.AWSEBSInTreePluginName, "good-pvc-vol-3"),
+			pvc:             getFakePersistentVolumeClaim("good-pvc", "vol-3", "resizable2", "good-pvc-vol-3"),
+			storageClass:    getFakeStorageClass("resizable2", csitranslationplugins.AWSEBSInTreePluginName),
+			pvcKey:          "default/good-pvc",
+			expansionCalled: true,
+		},
+		{
+			name:                "when csi migration is enabled for a in-tree plugin",
+			csiMigrationEnabled: true,
+			pv:                  getFakePersistentVolume("vol-4", csitranslationplugins.AWSEBSInTreePluginName, "csi-pvc-vol-4"),
+			pvc:                 getFakePersistentVolumeClaim("csi-pvc", "vol-4", "resizable3", "csi-pvc-vol-4"),
+			storageClass:        getFakeStorageClass("resizable3", csitranslationplugins.AWSEBSInTreePluginName),
+			pvcKey:              "default/csi-pvc",
+		},
+	}
+
+	for _, tc := range tests {
+		test := tc
+		if tc.pv != nil {
+			informerFactory.Core().V1().PersistentVolumes().Informer().GetIndexer().Add(tc.pv)
+		}
+
+		if tc.pvc != nil {
+			informerFactory.Core().V1().PersistentVolumeClaims().Informer().GetIndexer().Add(tc.pvc)
+		}
+		allPlugins := []volume.VolumePlugin{}
+		allPlugins = append(allPlugins, awsebs.ProbeVolumePlugins()...)
+		if tc.storageClass != nil {
+			informerFactory.Storage().V1().StorageClasses().Informer().GetIndexer().Add(tc.storageClass)
+		}
+		expc, err := NewExpandController(fakeKubeClient, pvcInformer, pvInformer, storageClassInformer, nil, allPlugins)
+		if err != nil {
+			t.Fatalf("error creating expand controller : %v", err)
+		}
+
+		if test.csiMigrationEnabled {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigration, true)()
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigrationAWS, true)()
+		}
+
+		var expController *expandController
+		expController, _ = expc.(*expandController)
+		var expansionCalled bool
+		expController.operationGenerator = operationexecutor.NewFakeOgCounter(func() (error, error) {
+			expansionCalled = true
+			return nil, nil
+		})
+
+		err = expController.syncHandler(test.pvcKey)
+		if err != nil && !test.hasError {
```

Comment:
Please also check for `err == nil && test.hasError` - test expected an error but got success.